### DISCUSSION TITLE
Main.js bundle not found (404)

### DISCUSSION THREAD

**Discussion by ptr1120** [keycloakify/keycloakify#98] [Q&A] [answered]
Hello,

I am trying to create a custom login theme using `keycloakify`. 
When running `react-scripts start` everything is fine and my custom css is correctly applied. But when building the theme with `build-keycloak-theme` and injecting it into the keycloak container (17.0.1) using `start_keycloak_testing_container.sh` the bundled javascript file is not found anymore and I see an empty login page due to the error `Failed to load resource: the server responded with a status of 404 (Not Found) main.362b1b93.js:1`.
Any ideas on how to solve this?

login.ftl contains:
```html
<script defer="defer" src="/static/js/main.362b1b93.js"></script>
```

![image](https://user-images.githubusercontent.com/582627/166231653-1c4801ee-4e3f-48b9-a6e7-2f0db632208c.png)

Best regards
Peter

**Comment by garronej** [comment]
Make sure you have run `yarn keycloak` (I forget sometimes)  

Could you please show me the url that gives a 404?

**Comment by ptr1120** [comment]
thanks for your answer @garronej . `keycloak` `(react-scripts build && build-keycloak-theme)` has run, I ensured once again.

This is the Url called when using "https://www.keycloak.org/app/" 
`http://localhost:8080/realms/myrealm/protocol/openid-connect/auth?client_id=myclient&redirect_uri=https%3A%2F%2Fwww.keycloak.org%2Fapp%2F%23url%3Dhttp%3A%2F%2Flocalhost%3A8080%26realm%3Dmyrealm%26client%3Dmyclient&state=97a05edd-8708-4751-8574-719eb00848b0&response_mode=fragment&response_type=code&scope=openid&nonce=99a522c4-420c-4367-9ea8-f3599c1b8780`

![image](https://user-images.githubusercontent.com/582627/166243762-e2bbce2c-290d-47d5-acd7-a7336af02bc8.png)

Any idea?

**Comment by garronej** [comment]
I can't see on your screenshot.  
I need the full URI of the asset that is giving a 404.  
It should be something like:  
http://localhost:8080/xxxx/yyy/zzzz/main.326xxxx.js

**Comment by ptr1120** [comment]
Ok, this is called in order to get the asset: `http://localhost:8080/static/js/main.362b1b93.js` (as defined in login.ftl by `src="/static/js/main.362b1b93.js"`)

![image](https://user-images.githubusercontent.com/582627/166247939-4aab300c-cfd4-4d24-96da-e0011e994a42.png)

**Comment by garronej** [comment, marked as answer]
I don't know exactly what you are doing...  
The import part hasn't be fixed by Keycloakify at all.  
This is how it should work:  
https://filedn.com/lYPEPucNMheV19aDu23lK0Q/misc/Screen%20Recording%202022-05-02%20at%2016.25.17.mov

**Comment by ptr1120** [comment]
Thanks for the detailed video. 
The part with the `${url.resourcesPath}` is completely missing in my build_keycloak/src/main/resources/theme/my-theme/login/resources/*.ftl files. The ressources are referenced absolute like
`<script defer="defer" src="/static/js/main.362b1b93.js">`

**Comment by garronej** [comment]
Well... I'll need more details then.  
Even beter would be the a repo where I can reproduce
